lib/spack/spack/test/spec_semantics.py

```diff
@@ -743,3 +743,15 @@ def test_spec_formatting(self):
             expected = getattr(arch, prop, "")
             actual = spec.format(named_str)
             assert str(expected) == actual
+
+    @pytest.mark.regression('9908')
+    def test_spec_flags_maintain_order(self):
+        spec_str = 'libelf %gcc@4.7.2 os=redhat6'
+        hashes = set(
+            Spec(spec_str).concretized().dag_hash() for _ in range(100)
+        )
+
+        # The hash has been precomputed and should be the same
+        # on every architecture.
+        assert len(hashes) == 1
```

Comment:
@alalazo everything else looks good but IMO this test is fragile. For example if the hashing algorithm changes then this test will probably have to change too. I think it would be best to retrieve the flags from each of the concretized Specs and ensure that it matches in each case (that would also help clarify that this test is intended to check compiler flags specifically).

I also think it's worth adding a comment on the issue (e.g. "Spack was assembling flags in a manner that could in different orderings for repeated concretizations of the same spec and config"). Someone reading this could track down #9908, but I think it's worth saving them some time.

Reply:
@alalazo Appears to be working - I can confirm with python 3 on ubuntu 18.04 that it now installs once and tells me 9 times that it's already installed which addresses my immediate issue thank you.

Reply:
Should be done